snuba/subscriptions/scheduler.py

```diff
@@ -67,12 +73,25 @@ def __get_subscriptions(self, current_time: datetime) -> List[Subscription]:
                 for uuid, data in self.__store.all()
             ]
             self.__last_refresh = current_time
+            self.__metrics.gauge(
+                "schedule.size",
+                len(self.__subscriptions),
+                tags={"partition": str(self.__partition_id)},
+            )
```

Comment:
Would it be worth measuring this closer to where you actually yield the ScheduledTask (lines 99-102).
I fear that, should the scheduling logic change below in a way that filters tasks, it would be easy to miss that the metric would have to change accordingly.

Reply:
I guess that I don't really anticipate that use case, and this is intended more just to get a sense of how many subscriptions exist in the store, so I think it is OK here near the data access bits. If we later add the ability to coalesce subscriptions or filter them somehow so that there are less subscriptions active than the number in the storage, I think that is a different statistic (e.g. effective size vs actual size).